From 3c79843de5aadceb032cb7a2fd339b3f5ddfa446 Mon Sep 17 00:00:00 2001
From: Yiqing Zeng <zack.zeng@rock-chips.com>
Date: Mon, 10 Aug 2020 20:00:57 +0800
Subject: [PATCH] media: i2c: ov2718: group hold launch immediately when
 single_start set

cause flicker occurs when launch after SOF in first few frames

Signed-off-by: Yiqing Zeng <zack.zeng@rock-chips.com>
Change-Id: I01c503d6376c9787bdaf20aa6d861e742458c8f9
---
 drivers/media/i2c/ov2718.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/ov2718.c b/drivers/media/i2c/ov2718.c
index d9f30a0c9004..edf167136f26 100644
--- a/drivers/media/i2c/ov2718.c
+++ b/drivers/media/i2c/ov2718.c
@@ -7,6 +7,13 @@
  * V0.0X01.0X01 add poweron function.
  * V0.0X01.0X02 fix mclk issue when probe multiple camera.
  * V0.0X01.0X03 add enum_frame_interval function.
+ * V0.0X01.0X04 update a new version
+ *	1.update the new get regulator method
+ *	2.correct initialization sequence
+ *	3.fix the level setting of reset gpio
+ *	4.support raw12bit linear/hdr mode
+ *	5.implement RKMODULE_SET/GET_HDR_CFG
+ * V0.0X01.0X05 group hold launch immediately when set hdr ae.
  */
 
 #include <linux/clk.h>
@@ -35,7 +42,7 @@
 #include <linux/mfd/syscon.h>
 #include <linux/rk-preisp.h>
 
-#define DRIVER_VERSION			KERNEL_VERSION(0, 0x01, 0x04)
+#define DRIVER_VERSION			KERNEL_VERSION(0, 0x01, 0x05)
 
 #ifndef V4L2_CID_DIGITAL_GAIN
 #define V4L2_CID_DIGITAL_GAIN		V4L2_CID_GAIN
@@ -8088,7 +8095,7 @@ static long ov2718_ioctl(struct v4l2_subdev *sd, unsigned int cmd, void *arg)
 		ret |= ov2718_write_reg(ov2718->client,
 					OV2718_REG_GROUP_CTL,
 					OV2718_REG_VALUE_08BIT,
-					0x10);
+					0x40);
 
 		/* single start */
 		ret |= ov2718_write_reg(ov2718->client,
-- 
2.35.3

